omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -195,43 +204,55 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					}
 					
 					if (Context.hasPrivilege(PrivilegeConstants.GET_OBS)) {
-						List<Obs> patientObs = Context.getObsService().getObservationsByPerson(p);
-						model.put("patientObs", patientObs);
-						Obs latestWeight = null;
-						Obs latestHeight = null;
+						Integer limit = getLimitParameter(request, as, "dashboard.observations.maximumNumberToShow");
+						Integer startIndex = getStartIndexParameter(request);
+						
+						Person person = (Person) p;
+						List<Person> persons = Collections.singletonList(person);
+						
+						// Get paginated observations for display
+						List<Obs> paginatedObs = Context.getObsService().getObservations(persons, null, null, null, null,
+						    null, Collections.singletonList("obsDatetime desc"), limit, startIndex, null, null, false);
+						
+						model.put("patientObs", paginatedObs);
+						
+						// Handle BMI calculation
 						String bmiAsString = "?";
+						
 						try {
 							String weightString = as.getGlobalProperty("concept.weight");
+							String heightString = as.getGlobalProperty("concept.height");
+							
 							ConceptNumeric weightConcept = null;
+							ConceptNumeric heightConcept = null;
+							Obs latestWeight = null;
+							Obs latestHeight = null;
+							
 							if (StringUtils.hasLength(weightString)) {
 								weightConcept = cs.getConceptNumeric(GeneralUtils.getConcept(weightString).getConceptId());
+								List<Obs> weightObs = Context.getObsService().getObservations(persons, null,
+								    Collections.singletonList(weightConcept), null, null, null,
+								    Collections.singletonList("obsDatetime desc"), 1, null, null, null, false);
+								if (!weightObs.isEmpty()) {
+									latestWeight = weightObs.get(0);
+								}
 							}
-							String heightString = as.getGlobalProperty("concept.height");
-							ConceptNumeric heightConcept = null;
+							
 							if (StringUtils.hasLength(heightString)) {
 								heightConcept = cs.getConceptNumeric(GeneralUtils.getConcept(heightString).getConceptId());
-							}
-							for (Obs obs : patientObs) {
-								if (obs.getConcept().equals(weightConcept)) {
-									if (latestWeight == null
-									        || obs.getObsDatetime().compareTo(latestWeight.getObsDatetime()) > 0) {
-										latestWeight = obs;
-									}
-								} else if (obs.getConcept().equals(heightConcept)
-								        && (latestHeight == null || obs.getObsDatetime().compareTo(
-								            latestHeight.getObsDatetime()) > 0)) {
-									latestHeight = obs;
+								List<Obs> heightObs = Context.getObsService().getObservations(persons, null,
+								    Collections.singletonList(heightConcept), null, null, null,
+								    Collections.singletonList("obsDatetime desc"), 1, null, null, null, false);
+								if (!heightObs.isEmpty()) {
+									latestHeight = heightObs.get(0);
 								}
 							}
-							if (latestWeight != null) {
```

Comment:
According to the original code, latestHeight or latestWeight was put in the model as long as it was not null. Did you intentionally change it?

Reply:
Did you see the above comment?